sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1549,3 +1549,12 @@ def test_numeric_pairwise_distances_datatypes(metric, global_dtype, y_is_x):
     dist = pairwise_distances(X, Y, metric=metric, **params)
 
     assert_allclose(dist, expected_dist)
+
+
+def test_nan_euclidean_support():
```

Comment:
We can parametrize passing the function at each iteration. We should also check the output of the pairwise distance, for instance making sure that we don't have `nan` values.